src/AnalyticsEventReporter.ts

```diff
@@ -1,38 +1,71 @@
+import { EventPayload, PartialPayload } from './EventPayload';
+import { EventAPIResponse } from './EventAPIResponse';
+import { getOrSetupSessionId } from './setupSessionId';
+import { name, version } from '../package.json';
+import { post } from './post';
+import merge from './merge';
+import { setupRequestUrl } from './setupRequestUrl';
 
 /** Represents an reporter is responsible for reporting analytics events. */
 export class AnalyticsEventReporter implements AnalyticsEventService {
-
     private config: AnalyticsConfig;
     private payload?: EventPayload;
-
     /**
      * @param config - necessary analytics config: Must provide one and only
      * one of API Key or Bearer Token.
-     * 
+     *
      * @param payload - (optional) desired event values to report
      */
     constructor(config: AnalyticsConfig, payload?: EventPayload) {
-        const apiKeyIsSet = config.key !== undefined;
-        const bearerTokenIsSet = config.bearer !== undefined;
-        const configIsValid = (apiKeyIsSet || bearerTokenIsSet) && !(apiKeyIsSet && bearerTokenIsSet);
-        if (!configIsValid) {
-            throw new Error("Provide one and only one of API Key or Bearer Token.")
-        }
-        this.config = config;
-        this.payload = payload;
+      const apiKeyIsSet = config.key !== undefined;
+      const bearerTokenIsSet = config.bearer !== undefined;
+      const configIsValid = (apiKeyIsSet || bearerTokenIsSet) && !(apiKeyIsSet && bearerTokenIsSet);
+      if (!configIsValid) {
+        throw new Error('Provide one and only one of API Key or Bearer Token.');
+      }
+      this.config = config;
+      this.payload = payload;
     }
 
     with(payload: EventPayload): AnalyticsEventService {
-        const currentPayload = this.payload === undefined ? payload : merge(this.payload, payload);
-        return new AnalyticsEventReporter(this.config, currentPayload);
+      const currentPayload = this.payload === undefined ? payload : merge(this.payload, payload);
+      return new AnalyticsEventReporter(this.config, currentPayload);
     }
 
-    report(payload: object): Promise<EventAPIResponse> | boolean {
-        return false;
+    public async report(newPayload?: PartialPayload): Promise<boolean | EventAPIResponse> {
+      const finalPayload = (this.payload && newPayload) ? (merge(this.payload, newPayload))
+        : this.payload ?? newPayload ?? function(){throw Error('EventPayload is empty');}();
+
+      if (this.config.sessionTrackingEnabled) {
+        const sessionId = newPayload?.sessionId ?? getOrSetupSessionId();
+        finalPayload.sessionId = sessionId;
+      }
+
+      finalPayload.clientSdk = {
+        [name]: version,
+      };
+
+      finalPayload.authorization = this.config.key ?? this.config.bearer;
+
+      const res = await post(
+        setupRequestUrl(
+          this.config.env,
+          this.config.region),
+        finalPayload as EventPayload,
+        this.config.forceFetch ?? true);
```

Comment:
Ah you're correct. 

If forceFetch is true, use fetch
Otherwise if undefined - defaults to false, use fetch if browser is eligible else use beacon. 

To avoid doing a null check I do need to change the forceFetch post() parameter type to be `boolean | undefined` otherwise since forceFetch is an optional config element, I would need to do `?? false` as otherwise we get the type error boolean | undefined could not be cast to boolean.